pkg/roll/execute_test.go

```diff
@@ -507,6 +508,78 @@ func TestMigrationHooksAreInvoked(t *testing.T) {
 	})
 }
 
+func TestRawSQLURLOption(t *testing.T) {
+	t.Parallel()
+
+	testutils.WithConnectionString(t, func(connStr string) {
+		ctx := context.Background()
+
+		st, err := state.New(ctx, connStr, schema)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := st.Init(ctx); err != nil {
+			t.Fatal(err)
+		}
+
+		t.Cleanup(func() {
+			if err := st.Close(); err != nil {
+				t.Fatal(err)
+			}
+		})
+
+		db, err := sql.Open("postgres", connStr)
+		if err != nil {
+			t.Fatal(err)
+		}
```

Comment:
I think this test could be made shorter by moving more of this setup into `WithConnectionString`. `WithConnectionString` can create the state. The signature of the callback could be: `fn func(*state.State, *sql.DB, connStr string)`